### PR TITLE
fix object ghost follows invisible cursor when scroll with right-click

### PR DESCRIPTION
**Pull request by guljam**
Fix issue https://github.com/OpenRCT2/OpenRCT2/issues/17221
Object ghost and tool tips following invisible cursor when scrolling with right click (like vanilla RCT2)

Fix List

    Placing attraction or track
    Footpath tool
    Land rights tool
    Guest and staff pickup
    Placing park entrance and guest start point in the scenario editor
    Staff patrol area
    Select area in cut-away view tool
    Tile inspector

### CODE REVIEW

File: src/openrct2-ui/input/MouseInput.cpp [hunk 1]
```diff
@@ -1175,13 +1175,12 @@ void ProcessMouseTool(const ScreenCoordsXY& screenCoords)
...
         if (w == nullptr)
             tool_cancel();
-        else
+        else if (input_get_state() != InputState::ViewportRight)
             window_event_tool_update_call(w, gCurrentToolWidget.widget_index, screenCoords);
     }
 }
 
-/**
- *
```
Comment (Broxzier) [inline src/openrct2-ui/input/MouseInput.cpp:1184]: Revert this change.

File: src/openrct2-ui/windows/PatrolArea.cpp [hunk 2]
```diff
@@ -151,6 +151,7 @@ class PatrolAreaWindow final : public Window
         auto mapTile = GetBestCoordsFromPos(screenCoords);
         if (!mapTile)
             return;
+
```
Comment (Broxzier) [inline src/openrct2-ui/windows/PatrolArea.cpp:154]: Revert this change.

**Review by Broxzier — CHANGES_REQUESTED** [review]
I hope this is the last one...

By reverting, I mean restoring the lines like how they are on develop, not using `git revert` because that adds a commit that undoes a previous commit.

File: src/openrct2-ui/input/MouseInput.cpp [hunk 3]
```diff
@@ -1175,13 +1175,12 @@ void ProcessMouseTool(const ScreenCoordsXY& screenCoords)
...
         if (w == nullptr)
             tool_cancel();
-        else
+        else if (input_get_state() != InputState::ViewportRight)
             window_event_tool_update_call(w, gCurrentToolWidget.widget_index, screenCoords);
     }
 }
 
 /**
- *
```
Comment (Broxzier) [inline src/openrct2-ui/input/MouseInput.cpp:1184]: One more. I'll fix this for you so we can get this merged.

**Review by Broxzier — APPROVED** [review]

**Comment by guljam** [comment]
@Broxzier 
Thanks for merging.
I wonder what you mean by "revert".
Does this mean command revert in git?
Or does it simply mean changing the file back to original with a text editor even when a commit occurs?

A few days ago I was a little confused when I heard you said to "revert" a language file.

Is it ok if I edit the wrong file, there are a lot of changes in the commit history but actually the file doesn't have any effect?
If not, do I have to delete the contents of the history with the reset command?

I think you're sensitive to file changes when you do code review.
So I created a new PR.

**Comment by Broxzier** [comment]
A branch should only contain the changes that belong to the change, any other changes should not be present, hence reverted to how things were. When an entire commit is faulty, you can use `git revert <commit hash>` to revert that commit, though even better is to rebase and remove the faulty commit from the branch.
